src/main/java/internetshop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,79 @@
+package internetshop.web.filters;
+
+import internetshop.lib.Injector;
+import internetshop.model.Role;
+import internetshop.model.User;
+import internetshop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector INJECTOR = Injector.getInstance("internetshop");
+    private static final UserService userService =
+            (UserService) INJECTOR.getInstance(UserService.class);
+    private static final String USER_ID = "user_id";
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/adminPage", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/orderInfo", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/completeOrder", List.of(Role.RoleName.USER));
+        protectedUrls.put("/products/all", List.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String requestUrl = req.getServletPath();
+
+        if (protectedUrls.get(requestUrl) == null) {
+            chain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        User user = userService.getById(userId);
+        if (isHasRights(user, protectedUrls.get(requestUrl))) {
+            chain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/views/accessDenied.jsp").forward(req, resp);
+            return;
+        }
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    private boolean isHasRights(User user, List<Role.RoleName> authorizedRoles) {
```

Comment:
```suggestion
    private boolean isHasRights(User user, List<Role.RoleName> authorizedRoles) {
```
isHasRights трішки каша з точки зору англ мови, давай зробимо неймінг аналогічний до прикладу з відео Богдана?